# Envirionment

In [ ]:
import os
import io
import time
import requests
import zipfile
import torch
import numpy as np

from dataset import acpds
from utils import roi_analysis as ra

In [ ]:
# Adjust CWD
directory = os.getcwd().replace('\\notebooks', '')
os.chdir(directory)

In [ ]:
# Download and store ACPDS
# Alternatively, you can manually download 'data' from README.md and upload 'data' under the 'dataset' folder to circumvent this if statement
if not os.path.exists('dataset/data'): 
    r = requests.get("https://pub-e8bbdcbe8f6243b2a9933704a9b1d8bc.r2.dev/parking%2Frois_gopro.zip")
    z = zipfile.ZipFile(io.BytesIO(r.content))
    z.extractall('dataset/data')

# Data

In [ ]:
train_ds, valid_ds, test_ds = acpds.create_datasets('dataset/data') # create training, validation, and test splits
image_batch, rois_batch, labels_batch = next(iter(valid_ds)) # retrieve next batch of data from validation set
image_raw, rois, labels = image_batch[0], rois_batch[0], labels_batch[0] # select first observation in batch

In [ ]:
# Calculate (avg. W, avg. H, W std., H std.) for each data set split

print(ra.calculate_avg_std_dimensions_for_dataset(train_ds))
print(ra.calculate_avg_std_dimensions_for_dataset(valid_ds))
print(ra.calculate_avg_std_dimensions_for_dataset(test_ds))

(0.11825147082472777, 0.12224295054808323, 0.050866807057034405, 0.07349692515270921)
(0.09843804208295685, 0.10224390168275152, 0.04948001700852599, 0.07446447972740446)
(0.10496666627349677, 0.10206117453398528, 0.05481679075294071, 0.08419208118209133)


# Results

### Training

- **avg. WIDTH:** 0.11825147082472777
- **WIDTH std.:** 0.050866807057034405
- **avg. HEIGHT:** 0.12224295054808323
- **HEIGHT std.:** 0.07349692515270921

- **min. RATIO:** 1.3823622176606862
- **max. RATIO:** 0.8639950203109152
- **avg. RATIO:** 0.9673479762599034

### Validation

- **avg. WIDTH:** 0.09843804208295685
- **WIDTH std.:** 0.04948001700852599
- **avg. HEIGHT:** 0.10224390168275152
- **HEIGHT std.:** 0.07446447972740446

- **min. RATIO:** 1.7623845864441137
- **max. RATIO:** 0.8370743815945649
- **avg. RATIO:** 0.9627766591732413

### Testing

- **avg. WIDTH:** 0.10496666627349677
- **WIDTH std.:** 0.05481679075294071
- **avg. HEIGHT:** 0.10206117453398528
- **HEIGHT std.:** 0.08419208118209133

- **min. RATIO:** 2.8065148316683155
- **max. RATIO:** 0.8578827597516495
- **avg. RATIO:** 1.0284681393564012

### **Overall (Average)**

- **avg. WIDTH:** 0.10721872639372713
- **WIDTH std.:** 0.051721204939500366
- **avg. HEIGHT:** 0.10884934225494002
- **HEIGHT std.:** 0.07738449535406833

- **min. RATIO:** 1.9837538785910385
- **max. RATIO:** 0.8529840538857099
- **avg. RATIO:** 0.9861975915965152